internal/oidc/token/token_handler_test.go

```diff
@@ -1199,12 +1209,100 @@ func TestRefreshGrant(t *testing.T) {
 					wantSuccessBodyFields:             []string{"refresh_token", "access_token", "id_token", "token_type", "expires_in", "scope"},
 					wantRequestedScopes:               []string{"openid", "offline_access"},
 					wantGrantedScopes:                 []string{"openid", "offline_access"},
+					wantGroups:                        goodGroups,
 					wantUpstreamRefreshCall:           happyOIDCUpstreamRefreshCall(),
 					wantUpstreamOIDCValidateTokenCall: happyUpstreamValidateTokenCall(refreshedUpstreamTokensWithRefreshTokenWithoutIDToken(), false),
 					wantCustomSessionDataStored:       upstreamOIDCCustomSessionDataWithNewRefreshToken(oidcUpstreamRefreshedRefreshToken),
 				},
 			},
 		},
+		{
```

Comment:
Maybe add a test for when the groups claim exists in the upstream refresh request but has an empty list as the value.

Reply:
Fixed!

Thanks for pointing that out because it also caused me to realize that any errors when parsing the upstream refresh groups claim would be returned in the wrong format and with the wrong status code. Added tests and new production code for that too.